### ISSUE TITLE
Block destructive syntax, or Block "phrases"

### ISSUE THREAD

**Issue by Nick-Lucas** [sqlfluff/sqlfluff#5988]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Consider the migration:

```sql
create or replace table important_table (
    insert_timestamp timestamp_ntz,
    name string,
);
```

What's wrong with this migration? Well you can't see it here but the table already exists, and the goal was just to add one new column, but the developer copied from the original migration and didn't consider that it's a destructive operation and will wipe the table. Instead an `alter` statement should have been used.

My team want to ensure that we catch issues which could lead to data-loss during development, and explicitely override/ignore with justification if needed.

We have tried using convention.blocked_words to achieve this, but neither blocked_words or blocked_regex have any support for whitespace, and so can't be used with "replace table" or "(replace\s+table)"

Ideally there would be a way to warn about dangerous operations and ensure these are surfaced before or during code-review

### Dialect

snowflake

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @Nick-Lucas - thanks for opening the issue.

As the documentation says:
> Only whole words can be blocked, not phrases, nor parts of words.

As you have found the closest built-in config you can use today is:

```ini
[sqlfluff:rules:convention.blocked_words]
blocked_words = REPLACE
```

That is broader than what you want, because it blocks any use of REPLACE, not just REPLACE TABLE.

If your team needs phrase-level blocking, the right fix is a small custom SQLFluff rule or plugin that evaluates the full statement text and flags REPLACE TABLE specifically. SQLFluff is already set up for custom rules, so that is feasible.
